src/RE/W/WorldSpaceMenu.cpp

```diff
@@ -0,0 +1,89 @@
+#include "RE/W/WorldSpaceMenu.h"
+#include "RE/I/InterfaceStrings.h"
+#include "RE/U/UI.h"
+#include "RE/U/UIMessage.h"
+
+namespace RE
+{
+	WorldSpaceMenu::~WorldSpaceMenu()
+	{
+		// TODO, unregister if registered on hudmodechangevent
+		if (menuNode.get() && menuNode.get()->parent) {
+			if (menuNode.get()->parent) {
+				menuNode.get()->parent->DetachChild2(menuNode.get());
+			}
+			menuNode->DeleteThis();
+			menuNode.reset();
+		}
+
+		menuName.~BSFixedString();
+		fxDelegate.reset();
+		uiMovie.reset();
+	}
+
+	WorldSpaceMenu::WorldSpaceMenu(bool a_registerHudModeChangeEvent, bool a_matchAsTopMenu, bool a_queueUpdateFixup)
+	{
+		this->registerForHudModeChangeEvent = a_registerHudModeChangeEvent;
+		this->matchAsTopMenu = a_matchAsTopMenu;
+		this->queueUpdateFixup = a_queueUpdateFixup;
+		this->Unk_09(RE::UI_MENU_Unk09::kNone);
+		this->menuFlags.set(RE::UI_MENU_FLAGS::kUsesCursor, RE::UI_MENU_FLAGS::kModal,
+			RE::UI_MENU_FLAGS::kDisablePauseMenu, RE::UI_MENU_FLAGS::kTopmostRenderedMenu,
+			RE::UI_MENU_FLAGS::kAllowSaving, RE::UI_MENU_FLAGS::kInventoryItemMenu);
```

Comment:
Were these defaults observed from the engine? Some of them seem specific to QuickLoot VR like the kInventoryItemMenu which might make more sense to be part of the QuickLoot VR mod. Other use cases for general world space menus would need to disable a lot of these (e.g. floating healthbar menu).

Reply:
I rechecked and no, those aren't the defaults and should be updated. The default is `*&a1->menuFlags._impl = 0x10900i64;` which translates to `kRequiresUpdate = 1 << 8,` `kAllowSaving = 1 << 11,` and `kAssignCursorToRenderer = 1 << 16,`